lambdas/authorizationcode/src/main/java/uk/gov/di/ipv/cri/passport/authorizationcode/AuthorizationCodeHandler.java

```diff
@@ -139,17 +140,24 @@ public APIGatewayProxyResponseEvent handleRequest(
                     HttpStatus.SC_OK, Map.of(AUTHORIZATION_CODE, authorizationCode));
         } catch (HttpResponseExceptionWithErrorBody e) {
             return ApiGatewayResponseGenerator.proxyJsonResponse(
-                    e.getStatusCode(), e.getErrorBody());
+                    e.getStatusCode(),
+                    new OAuthErrorResponse(
+                            OAuthErrorResponse.SERVER_ERROR_CODE, e.getErrorReason()));
```

Comment:
For the `server_error` code value we can get a public static value of this from a nimbus library as well.

`OAuth2Error.SERVER_ERROR_CODE`

Reply:
We can then call the toJSONObject() method on the Nimbus ErrorObject value for what we return in the body here. And this toJSONObject will put it in the correct format

Reply:
Done 👍